### PR TITLE
[admin] allow trial extension for orgs currently on trial

### PR DESCRIPTION
**Pull request by JerrySentry**
### Purpose/Motivation
What is the feature? Why is this being done?

Currently the admin app can extend a org trial, but only after they have expired and turned back to basic plan.
This change allows the orgs' trials to be extended while they're still trialing.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. In 2022 this entity acquired Codecov and as result Sentry is going to need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: codecov_auth/tests/test_admin.py [hunk 1]
```diff
@@ -293,6 +297,26 @@ def test_start_trial_action(self, mock_start_trial_service):
...
 
+    @patch("plan.service.PlanService._start_trial_helper")
+    def test_extend_trial_action(self, mock_start_trial_service):
+        mock_start_trial_service.return_value = None
+        org_to_be_trialed = OwnerFactory()
+        org_to_be_trialed.plan = PlanName.TRIAL_PLAN_NAME.value
+        org_to_be_trialed.save()
+
+        res = self.client.post(
+            reverse("admin:codecov_auth_owner_changelist"),
```
Comment (ajay-sentry) [inline codecov_auth/tests/test_admin.py:308]: looked this up, honestly a pretty interesting django feature lol

File: plan/service.py [hunk 2]
```diff
@@ -139,21 +140,28 @@ def available_plans(self, owner: Owner) -> List[PlanData]:
         return available_plans
 
     def _start_trial_helper(
-        self, current_owner: Owner, end_date: datetime = None
+        self,
+        current_owner: Owner,
+        end_date: datetime = None,
```
Comment (ajay-sentry) [inline plan/service.py:145]: This should be datetime | None for the type right?

**Review by ajay-sentry — APPROVED** [review]
makes sense, just 1 small nit

File: plan/service.py (hunk 2 above)
Comment (JerrySentry) [reply plan/service.py]: Yes, updated to `end_date: Optional[datetime] = None`

**Pull request merged**